FEATURE: Refactor to TYPO3 v10 compatibility

stephanschuler opened the pull request:
Several classes we hook into, xclass or rely on are gone now, work
differently or just introduced variable scopes. The single biggest
change is how PageLayoutView and PageLayoutController interact,
which brings a lot of smoall other changes.

Keepiong translations in connected mode is now done by TYPO3
automatically keeping l10n_mode=exclude fields in sync. Having both,
l10n_mode=exclude and l10n_display=defaultAsReadonly always on
rivides the same backend experience as having l10n_mode=exclude in FE
and l10n_display=readonly in backend plus manually hooking into the
data handler, so this hack is gone now and replaced by proper TCA.

The backend position service now works without ExtJS which is no longer
part of the TYPO3 backend. Instead it now relies on the requirejs being
present and proeprly configured.

stephanschuler committed:
FEATURE: Refactor to TYPO3 v10 compatibility

Several classes we hook into, xclass or rely on are gone now, work
differently or just introduced variable scopes. The single biggest
change is how PageLayoutView and PageLayoutController interact,
which brings a lot of smoall other changes.

Keepiong translations in connected mode is now done by TYPO3
automatically keeping l10n_mode=exclude fields in sync. Having both,
l10n_mode=exclude and l10n_display=defaultAsReadonly always on
rivides the same backend experience as having l10n_mode=exclude in FE
and l10n_display=readonly in backend plus manually hooking into the
data handler, so this hack is gone now and replaced by proper TCA.

The backend position service now works without ExtJS which is no longer
part of the TYPO3 backend. Instead it now relies on the requirejs being
present and proeprly configured.

liwo commented on Classes/Hooks/DataHandler/SkipIndividualShortcutsForDifferentLanguages.php:
```diff
@@ -0,0 +1,48 @@
+{
+    public function processDatamap_beforeStart(DataHandler $dataHandler): void
+    {
+        if (!GeneralUtility::_GET('DDcopy') && !GeneralUtility::_GET('reference')) {
```
Willst du hier wirklich nur raus wenn sowohl `DDcopy` als auch `reference` nicht gesetzt sind? Oder sollte das eher ein `||` werden?

stephanschuler replied:
Diese Condition war damals wohl richtig weil genau dieser Zustand der Indikator dafür war, dass die Referenz durch gridlements erzeugt wurde und nicht durch den Core.
Aktuell können wir aber die Condition komplett verzichten weil der zugehörige Teil aus gridelements mit Version 7 und TYPO3 ">=7.6.0" komplett verschwunden ist. Das heißt der Code ist schon im aktuellen condensedbelayout master den wir mit TYPO3 "^8.6" angeben überflüssig.

Done.

liwo commented on Classes/Xclass/Gridelements/Backend/ItemsProcFuncs/CTypeList.php:
```diff
@@ -17,75 +17,74 @@
+
+            $languageParent = BackendUtility::getRecord('tt_content', $row['l18n_parent']);
+            if ($languageParent) {
+                \xdebug_break();
```
Hust

stephanschuler replied:
Bitte gehen Sie weiter, hier gibt es nichts zu sehen.

stephanschuler committed:
TASK: Remove obsolete hool condition

This data is gone since TYPO3 v7.6 and gridelements v7.1. There is no
point in keeping it for condensed be layout that requires TYPO3 v10
and gridelements v10.

stephanschuler committed:
TASK: Add extension key to composer extras
